pyodide_build/mkpkg.py

```diff
@@ -245,18 +245,46 @@ def update_package(
         old_fmt = "sdist"
 
     pypi_metadata = _get_metadata(package, version)
+
+    # Grab versions from metadata
     pypi_ver = Version(pypi_metadata["info"]["version"])
     local_ver = Version(yaml_content["package"]["version"])
-    already_up_to_date = pypi_ver <= local_ver and (
+
+    # and grab checksums from metadata
+    source_fmt = source_fmt or old_fmt
+    dist_metadata = _find_dist(pypi_metadata, [source_fmt])
+    sha256 = dist_metadata["digests"]["sha256"]
+    sha256_local = yaml_content["source"].get("sha256")
+
+    # fail if local version is newer than PyPI version
+    # since updating isn't possible in that case
+    if pypi_ver <= local_ver:
+        raise MkpkgFailedException(
+            f"Local version {local_ver} is newer than PyPI version {pypi_ver}, "
+            f"cannot update {package}. Please verify in case the version was "
+            "updated manually and is correct."
+        )
+
+    # conditions to check if the package is up to date
+    is_sha256_up_to_date = sha256 == sha256_local
+    is_version_up_to_date = pypi_ver <= local_ver
```

Comment:
When `pypi_ver <= local_ver` is true, it fails with exception, but `is_version_up_to_date` checks the same condition after the exception.

Reply:
Oops! Thanks for noticing. 143b729360f48cc912b634ffcc1c72c7ffd2463a fixes this – the version comparison should be made in equality (`pypi_ver == local_ver`), and the impossible update scenario should have a strict greater-than (`pypi_ver < local_ver`).